dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackerAccessManager.java

```diff
@@ -807,4 +803,20 @@ private boolean isNull( ProgramStage programStage )
         return programStage == null || programStage.getProgram() == null;
     }
 
+    private boolean isOrgUnitAccessible( User user, Program program, OrganisationUnit orgUnit )
+    {
+        // always allow if these are null. Internal process?
```

Comment:
What do you mean by `Internal process?`

This logic looks like it is the same for DefaultTrackerAccessManager and AbstractEventService? Code duplicated in 2 places is not per se a bad thing if they do not mean exactly the same thing. If they are likely to evolve differently. If the logic in DefaultTrackerAccessManager and AbstractEventService isOrgUnitAccessible should stay in sync then it is dangerous to have it in two places. They could get out of sync if one forgets to update both. Which one is it? 1. Looks the same but has different meaning 2. looks and means the same thing and should stay in sync 😄

Reply:
yes, the accessible check should be centralized.